src/utils/s3_mods.py

```diff
@@ -0,0 +1,148 @@
+# General functions
+import json
+import pandas as pd
+from io import StringIO
+import logging
+from typing import List
+import subprocess
+import os
+
+# Functions specific to s3 bucket
+import boto3
+import raz_client
+from rdsa_utils.cdp.helpers.s3_utils import file_exists, create_folder_on_s3
+
+
+# set up logging
+s3_logger = logging.getLogger(__name__)
+
+
+def create_client(config):
+    client = boto3.client("s3")
+    raz_client.configure_ranger_raz(client, ssl_file=config["s3"]["ssl_file"])
+    return client
+
+
+# Read a CSV file into a Pandas dataframe
+def rd_read_csv(filepath: str, **kwargs) -> pd.DataFrame:
+    """Reads a csv from s3 bucket into a Pandas Dataframe using pydoop. 
+    If "thousands" argument is not specified, sets it to ",". 
+    Allows to use any additional keyword arguments of Pandas read_csv method.
+
+    Args:
+        filepath (str): Filepath (Specified in config)
+        kwargs: Optional dictionary of Pandas read_csv arguments
+    Returns:
+        pd.DataFrame: Dataframe created from csv
+    """
+    # Open the boto3 client
+    s3_client = config["client"]
+    with s3_client.get_object(
+        Bucket=config["s3"]["s3_bucket"],
+        Key=filepath
+    )['Body'] as file:
+        
+        # If "thousands" argument is not specified, set it to ","
+        if "thousands" not in kwargs:
+            kwargs["thousands"] = ","
+
+        # Read the scv file using the path and keyword arguments
+        try:
+            df = pd.read_csv(file, **kwargs)
+        except Exception:
+            if "usecols" in kwargs:
+                rd_logger.info("Columns not found: " + str(kwargs["usecols"]))
+            rd_logger.error(f"Could not read specified file: {filepath}")
+
+            raise ValueError
+    return df
+  
+
+def rd_write_csv(filepath: str, data: pd.DataFrame) -> None:
+    """Writes a Pandas Dataframe to csv in s3 bucket
+
+    Args:
+        filepath (str): Filepath (Specified in config)
```

Comment:
The filepath won't necessarily always be specified in the config. Also, the function directly reads in the string, making this irrelevant to the context of the function.

Reply:
Fixed this. resolving